### PR TITLE
Add pkg to kickstart via API

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr Hracek <phracek@redhat.com>

### CODE REVIEW

File: preup/script_api.py [hunk 1]
```diff
@@ -532,6 +532,16 @@ def print_home_dirs(user_name=""):
         pass
 
 
+def add_pkg_to_kickstart(pkg_name):
+    tmp = get_file_content(SPECIAL_PKG_LIST, "rb", True)
+    for pkg in pkg_name.split():
+        tmp.append(pkg)
+
+    write_to_file(SPECIAL_PKG_LIST, "wb", True)
```
Comment (pirat89) [inline preup/script_api.py]: missing parameter

**Comment by pirat89** [comment]
non functional.
Wouldn't be just append of new pkgs to the end of the file more effective then:
read file -> append pkgs -> write all back to the file?

**Comment by phracek** [comment]
Well, I have appended packages to a file.

**Comment by phracek** [comment]
@pirat89 Can you please review it again? What do you think about it?

File: preup/script_api.py [hunk 2]
```diff
@@ -532,6 +532,12 @@ def print_home_dirs(user_name=""):
         pass
 
 
+def add_pkg_to_kickstart(pkg_name):
+    for pkg in pkg_name.split():
+        write_to_file(SPECIAL_PKG_LIST, "a+b", pkg)
+    return 0
+
+
```
Comment (pirat89) [inline preup/script_api.py]: what about:
```python
def add_pkg_to_kickstart(pkgs):
    empty = False
    if(!isinstanceof(pkgs, list)):
        # string - pkgs delimited by whitespace
        if(len(pkgs.strip()) == 0):
            empty = True
        else:
           # make list from string
            pkgs = pkgs.strip().split()
    else:
        # list of packages
        if(len(pkgs) == 0):
            empty = True
    if(empty is True):
        log_debug("Missing parameters! Any package will be added.")
        return 1
    write_to_file(SPECIAL_PKG_LIST, "a+b", pkgs)
    return 0
```

It accepts list or string. I count that developer will use more list than strings in this case. Function write_to_file accept list & string too. It's untested, so maybe there are mistakes. Just written here as comment.

File: common.sh [hunk 3]
```diff
@@ -427,6 +428,18 @@ print_home_dirs() {
   conf_get_section "$_uconf_file" "home-dirs"
 }
 
+#Function adds a package to special_pkg_list
+add_pkg_to_kickstart() {
+  [ $# -eq 0  ] && {
+    echo "Missing parameters! Any package will be added." >&2
```
Comment (pirat89) [inline common.sh]: May log_debug will be better in this case. Not sure about that. What do You think?
```bash
log_debug "Missing parameters! Any package will be added."
```

**Pull request merged**